cmd/kube-scheduler/app/options/deprecated.go

```diff
@@ -87,16 +87,22 @@ func (o *DeprecatedOptions) Validate() []error {
 	return errs
 }
 
-// ApplyTo sets cfg.AlgorithmSource from flags passed on the command line in the following precedence order:
+// PolicyConfig returns whether a legacy policy config has been set
+func (o *DeprecatedOptions) PolicyConfig() bool {
+	if o == nil {
+		return false
+	}
+	return o.UseLegacyPolicyConfig || len(o.PolicyConfigFile) > 0 || len(o.PolicyConfigMapName) > 0 || len(o.AlgorithmProvider) > 0
+}
+
+// ApplyAlgorithmSourceTo sets cfg.AlgorithmSource from flags passed on the command line in the following precedence order:
 //
 // 1. --use-legacy-policy-config to use a policy file.
 // 2. --policy-configmap to use a policy config map value.
 // 3. --algorithm-provider to use a named algorithm provider.
-//
-// This function is only called when no config file is provided.
-func (o *DeprecatedOptions) ApplyTo(cfg *kubeschedulerconfig.KubeSchedulerConfiguration) error {
+func (o *DeprecatedOptions) ApplyAlgorithmSourceTo(cfg *kubeschedulerconfig.KubeSchedulerConfiguration) {
```

Comment:
I think I prefer we extract `ApplyAlgorithmSource`, but leave an `ApplyTo` function which calls it.

Reply:
done